scripts/pylint_custom_plugin/pylint_guidelines_checker.py

```diff
@@ -890,10 +891,14 @@ def visit_functiondef(self, node):
                     try:
                         # infer_call_result gives the method return value as a string
                         returns = next(node.infer_call_result()).as_string()
```

Comment:
just a small note - I remember this not being a super reliable way to get the return type for client method. If there is a better way that you've found, we should reconsider :)

Reply:
I added in a check after calling infer_call_result to see if it was inferred or not and if it isn't inferred, then I look for the astroid.Return in the body of the function node and grab the name of what is being returned